sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -54,6 +54,9 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   @transient
   final val sqlContext = SparkSession.getActiveSession.map(_.sqlContext).orNull
 
+  // whether we should fallback when hitting compilation errors caused by codegen
+  private val codeGenFallBack = sqlContext == null || sqlContext.conf.codegenFallback
```

Comment:
Is it better to add `!Utils.isTesting &&`?

Reply:
Originally, I did it like what you said. However, I need to remove the [test case](https://github.com/gatorsmile/spark/blob/9f3a1cd0c2e95bd6239facf3584d590a88aafce4/sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala#L2006). Then, I think we can keep using the `codegenFallback` for controlling it.

Reply:
I see